src/kmeans.jl

```diff
@@ -5,7 +5,7 @@
 mutable struct KmeansResult{T<:AbstractFloat} <: ClusteringResult
```

Comment:
Since `kmeans()` functions now accept `AbstractMatrix{<:Real}`, I guess here it also should be relaxed

```suggestion
mutable struct KmeansResult{T<:Real} <: ClusteringResult
```